src/ol/source/UrlTile.js

```diff
@@ -188,9 +178,7 @@ class UrlTile extends TileSource {
   setUrls(urls) {
     this.urls = urls;
     const key = urls.join('\n');
-    this.setTileUrlFunction(this.fixedTileUrlFunction ?
-      this.fixedTileUrlFunction.bind(this) :
-      createFromTemplates(urls, this.tileGrid), key);
+    this.setTileUrlFunction(createFromTemplates(urls, this.tileGrid), key);
   }
 
```

Comment:
This is what I meant by "remove that logic". However, I'm not sure if we were on the same page since this is technically a breaking change.

Reply:
This is not exactly what I had in mind. I've created #8765, which replaces this pull request and #8750. Please let me know what you think.